File: client/lib/core/models/event.dart
class Event {
  final int id;
  final String name;
  final String description;
  final String place;
  final String dateStart;
  final String dateEnd;
  final bool transportActive;
  final String transportStart;

  Event({
    required this.id,
    required this.name,
    required this.description,
    required this.place,
    required this.dateStart,
    required this.dateEnd,
    required this.transportActive,
    required this.transportStart,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      place: json['place'] ?? '',
      dateStart: json['date_start'] ?? '',
      dateEnd: json['date_end'] ?? '',
      transportActive: json['transport_active'] ?? false,
      transportStart: json['transport_start'] ?? 'Indefini',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'place': place,
      'date_start': dateStart,
      'date_end': dateEnd,
      'transport_active': transportActive,
      'transport_start': transportStart,
    };
  }
}

File: backend/internal/models/event.go
package models

import "gorm.io/datatypes"

// Event représente la structure d'un événement
type Event struct {
	ID              uint           `gorm:"primaryKey" json:"id"`
	Name            string         `json:"name" required:""`
	Description     string         `json:"description" required:""`
	Place           string         `json:"place"`
	DateStart       string         `json:"date_start"`
	DateEnd         string         `json:"date_end"`
	TransportActive bool           `json:"transport_active" gorm:"default:false"`
	Activities      datatypes.JSON `json:"activities"` // Utilisation de JSON pour les activités
	TransportStart  string         `json:"transport_start"`
	CreatedBy       uint           `json:"created_by"` // ID de l'utilisateur qui a créé l'événement
	Cagnotte        float64        `json:"cagnotte" gorm:"default:0"`
}

type EventTransportUpdate struct {
	TransportActive bool `json:"transport_active"`
}
